### ISSUE TITLE
Support Sequelize : Must implement cross joins (carthesian products)

### ISSUE THREAD

**Issue by oguimbal** [oguimbal/pg-mem#26]
Sequelize is generating requests using "comma-join-syntax", which is currently not supported.


```
👉 Failed query:

    SELECT i.relname AS name, ix.indisprimary AS primary, ix.indisunique AS unique, ix.indkey AS indkey, array_agg(a.attnum) as column_indexes, array_agg(a.attname) AS column_names, pg_get_indexdef(ix.indexrelid) AS definition FROM pg_class t, pg_class i, pg_index ix, pg_attribute a WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND a.attrelid = t.oid AND t.relkind = 'r' and t.relname = 'Users' GROUP BY i.relname, ix.indexrelid, ix.indisprimary, ix.indisunique, ix.indkey ORDER BY i.relname;;

💀 Syntax error at line 1 col 239:

  SELECT i.relname AS name, ix.indisprimary AS primary, ix.indisunique AS unique, ix.indkey AS indkey, array_agg(a.attnum) as column_indexes, array_agg(a.attname) AS column_names, pg_get_indexdef(ix.indexrelid) AS definition FROM pg_class t,
                                                                                                                                                                                                                                                ^
Unexpected comma token: ",". Instead, I was expecting to see one of the following:
```

This syntax is not supported as of today because it was kind of harder to implement, since it can lead to cross-joins (carthesian products).


 => This must be done.

Unit test stub for sequelize:  [src/tests/sequelize-real.spec.ts](https://github.com/oguimbal/pg-mem/blob/master/src/tests/sequelize-real.spec.ts)  (currently ignored)

**Comment by joeybenenati** [comment]
Any progress on this issue? I'd be interested in collaborating

**Comment by oguimbal** [comment]
Hi @joeybenenati , 

I dont have time to spend these days, so issue fixing is a bit on hold :(

You can try to help, but this definitely is a very hard issue, since it requires a significant refactoring of how joins work.
The current implementation is [here](https://github.com/oguimbal/pg-mem/blob/master/src/transforms/join.ts), but it is not a good implementation, and does not support cartesian joins

**Comment by oguimbal** [comment]
Hi ! This will be resolved soon thanks to the work of @janpaepke on the query parser.

In the meantime, here is a workaround that bypasses the introspection query that fails:

```typescript
   const db = newDb();
    db.public.interceptQueries(sql => {
        if (sql.includes(' FROM pg_class t, pg_class i')) {
            return []
        }
    });

    const sequelize = new Sequelize({
        dialect: 'postgres',
        dialectModule: mem.adapters.createPg(),
    });

    //  define your schema here
    
    await sequelize.sync(); // should work !
```

(it just forces to return an empty result for the incriminated query... which is OK given that we're on an empty database: there is nothing to introspect !)

**Comment by oguimbal** [comment]
Fixed in `pg-mem@2.4.0`  (should not even need the workaround above: pg-mem now supports cross-joins)
